### ISSUE TITLE
Tests for ConvolutionTranspose3D are probably invalid.

### ISSUE THREAD

**Issue by tsocha** [Microsoft/onnxruntime#699]
**Describe the bug**
Normally test data for convolution is in format:
- [N, C, H, W] or [N, H, W, C] for 2D
- [N, C, D, H, W] or [N, D, H, W, C] for 3D
but in these tests: https://github.com/Microsoft/onnxruntime/blob/master/onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc#L394-L474
data shape rank is 4 for both 2D and 3D tests.
3D and 2D tests differ only in rank of strides and amount of pads dimensions which seems invalid.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04.6 LTS
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: master
- Python version: n/a
- GCC/Compiler version (if compiling from source): 5.4.0
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

**Expected behavior**
Expect test to pass using different data shapes.

**Comment by pranavsharma** [comment]
Thanks for reporting this. Is your finding consistent with other frameworks? Or the onnx tests?

**Comment by tsocha** [comment]
Yes I checked onnx tests, pytorch and tensorflow.
In all of these cases they use 5D input for 3d convolutions and pooling.
In the onnxruntime tests I linked above padding might be invalid because there should be 6 zeros(above and below each axis) and in these tests there are 9 zeros.
Gathered data:
ONNX python unit tests input shapes:
convtranspose_3d: `[1, 1, 3, 4, 5]` and `[1, 2, 3, 3, 3]`
averagepool_3d: `[1, 3, 32, 32, 32]`
maxpool_3d: `[1, 3, 32, 32, 32]`
pytorch-converted.avg_pool_3d: `[2, 3, 4, 4, 4]`
pytorch-converted.conv_3d: `[2, 3, 3, 4, 5]`

pytorch conv3d: https://pytorch.org/docs/stable/nn.html#id22
pytorch conv-transpose3d: https://pytorch.org/docs/stable/nn.html#conv-transpose3d
tensorflow conv3d: https://www.tensorflow.org/api_docs/python/tf/nn/conv3d
tensorflow conv3d_transpose: https://www.tensorflow.org/api_docs/python/tf/nn/conv3d_transpose

**Comment by hariharans29** [comment]
@tsocha - Thanks for reporting this. Yes, that seems right. It doesn't look like a "3D" test case as the test name suggests. Let me add some real "3D" test cases :).

**Comment by hariharans29** [comment]
Hi @tsocha  - I created a PR to rename the misleading test names but I am unable to add 3D tests right away as they are not supported by the implementation yet. Will track that as a separate issue and will close this issue after the PR renaming the misleading test names is merged. Thanks for reporting this.
